src/border_agent/border_agent.cpp

```diff
@@ -327,6 +327,12 @@ void BorderAgent::HandleBorderAgentMeshCoPServiceChanged(bool aIsActive, uint16_
 
 #endif // OTBR_ENABLE_BORDER_AGENT_MESHCOP_SERVICE
 
+void BorderAgent::HandleThreadRoleChanged(otDeviceRole aRole)
+{
+    otbrLogInfo("Thread role changed to %s", otThreadDeviceRoleToString(aRole));
+    SetEnabled(aRole != OT_DEVICE_ROLE_DISABLED);
```

Comment:
I am not sure this is generally correct.  See https://github.com/openthread/ot-br-posix/issues/3068#issuecomment-3348114872